mvc/src/main/java/com/interface21/webmvc/servlet/mvc/tobe/HandlerAdapterRegistry.java

```diff
@@ -0,0 +1,39 @@
+package com.interface21.webmvc.servlet.mvc.tobe;
+
+import com.interface21.webmvc.servlet.ModelAndView;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class HandlerAdapterRegistry {
+	private final List<HandlerAdapter> handlerAdapters;
+
+	public HandlerAdapterRegistry(HandlerAdapter... handlerAdapters) {
+		this.handlerAdapters = Arrays.stream(handlerAdapters)
+				.toList();
+	}
+
+	public ModelAndView handle(Object handler, HttpServletRequest request, HttpServletResponse response) throws Exception {
+		HandlerAdapter handlerAdapter = getHandlerAdapter(handler);
+		return handlerAdapter.handle(request, response, handler);
+	}
+
+	private HandlerAdapter getHandlerAdapter(Object handler) {
+		return handlerAdapters.stream()
+				.filter(handlerAdapter -> handlerAdapter.supports(handler))
+				.findAny()
+				.orElseThrow(() -> new IllegalStateException("지원가능한 adapter가 없습니다."));
+	}
+
+	public void add(final HandlerAdapter handlerAdapter) {
+		handlerAdapters.stream()
+				.filter(adapter -> adapter.getClass().equals(handlerAdapter.getClass()))
+				.findAny()
+				.ifPresent(adapter -> {
+					throw new IllegalArgumentException("이미 등록된 handlerAdapter 입니다.");
+				});
```

Comment:
equal and hashcode 재정의를 통해 더 간단하게 구현해볼 수 있지 않을까요?
```suggestion
if(handlerAdapters.contains(handlerAdapater)) {
    throw new IllegalArgumentException("이미 등록된 어댑터입니다.");
}
```